dbt_coves/tasks/generate/docs.py

```diff
@@ -0,0 +1,165 @@
+        try:
+            with open(catalog_path, "r") as f:
+                return json.load(f)
+        except FileNotFoundError:
+            raise DbtCovesGenerateDocsException(f"Catalog.json not found at {catalog_path}")
+
+    def _merge_catalogs(
+        self, local_catalog: Dict[str, Any], target_catalog: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """
+        Merge nodes and sources from the state catalog.json into the local docs.
+        """
+        n_nodes_merged = sum(
+            1
+            for key in target_catalog.get("nodes", {})
+            if key not in local_catalog.get("nodes", {})
+        )
+        n_sources_merged = sum(
+            1
+            for key in target_catalog.get("sources", {})
+            if key not in local_catalog.get("sources", {})
+        )
+
+        local_catalog["nodes"].update(
+            (key, value)
+            for key, value in target_catalog.get("nodes", {}).items()
+            if key not in local_catalog.get("nodes", {})
+        )
+        local_catalog["sources"].update(
+            (key, value)
+            for key, value in target_catalog.get("sources", {}).items()
+            if key not in local_catalog.get("sources", {})
+        )
+
+        console.print(
+            f"Merged [green]{n_nodes_merged} nodes[/green] and [green]{n_sources_merged} sources[/green] into",
+            "your local catalog.json",
+        )
+
+    def _write_catalog_json(self, catalog: Dict[str, Any], docs_folder: Path):
+        """
+        Write the catalog.json to the docs_folder
+        """
+        catalog_path = Path(docs_folder, "catalog.json")
+        with open(catalog_path, "w") as f:
+            json.dump(catalog, f)
+
+    def _merge_dbt_catalogs(self, local_path: Path, state_path: Path):
+        """
+        Merge the catalog.json from the stsate into the local docs.
+        """
+        # Get the source and target catalogs
+        local_catalog = self._get_catalog_json(local_path)
+        target_catalog = self._get_catalog_json(state_path)
+
+        # Merge the catalogs
+        self._merge_catalogs(local_catalog, target_catalog)
+
+        # Write the merged catalog to the source
+        self._write_catalog_json(local_catalog, local_path)
+
+    @trackable
+    def run(self):
+        self._generate_dbt_docs()
+
+        local_docs_path = Path(self.config.project_root, "target")
+        self._fix_dbt_docs_links(local_docs_path)
+
+        merge_deferred = self.get_config_value("merge_deferred")
+        state_location = self.get_config_value("state")
+
+        if merge_deferred:
+            target_docs_path = Path(state_location, "catalog.json")
+            if not state_location or not target_docs_path.exists():
+                console.print(
+                    "A valid [red][i]--state[/i][/red] argument is required "
+                    "when using [yellow]--merge-deferred[/yellow]"
+                )
+                return -1
```

Comment:
negative errorcodes are not standard, see https://stackoverflow.com/questions/47265667/return-code-on-failure-positive-or-negative